crates/re_viewer_context/src/store_hub.rs

```diff
@@ -234,7 +254,34 @@ impl StoreHub {
             }
         }
 
-        self.active_application_id = Some(app_id);
+        if self.active_application_id.as_ref() == Some(&app_id) {
+            return;
+        }
+
+        self.active_application_id = Some(app_id.clone());
+        self.active_rec_id = None;
+
+        // Find any matching recording and activate it
```

Comment:
Seems like we should track active-recording-per-app so the last recording you selected will be stick. Also if it's the first time opening the app sorting these by last-received or latest-modify-time would be nice rather than choosing a random one bashed on the hashmap order.

Reply:
That would be nice, yes.

I'll see if I can fix the random order for now.